components/motor/gpio/motor_encoder_test.go

```diff
@@ -4,755 +4,279 @@ import (
 	"context"
 	"sync"
 	"testing"
-	"time"
 
 	"go.viam.com/test"
 	"go.viam.com/utils/testutils"
 
-	"go.viam.com/rdk/components/board"
-	fakeboard "go.viam.com/rdk/components/board/fake"
 	"go.viam.com/rdk/components/encoder"
-	"go.viam.com/rdk/components/encoder/incremental"
-	"go.viam.com/rdk/components/encoder/single"
 	"go.viam.com/rdk/components/motor"
-	fakemotor "go.viam.com/rdk/components/motor/fake"
 	"go.viam.com/rdk/logging"
-	"go.viam.com/rdk/operation"
 	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/testutils/inject"
 )
 
-func nowNanosTest() uint64 {
-	return uint64(time.Now().UnixNano())
-}
+const (
+	encoderName = "encoder"
+	motorName   = "motor"
+)
 
-type fakeDirectionAware struct {
-	m *fakemotor.Motor
+type stuff struct {
```

Comment:
[nit] change name to be descriptive. Maybe `injectedMotorState`?

Reply:
oops yes